lib/ansible/modules/network/aci/aci_aaa_user.py

```diff
@@ -0,0 +1,239 @@
+    description:
+    - The lifetime of the locally-authenticated user password.
+  aaa_password_update_required:
+    description:
+    - Whether this account needs password update.
+    type: bool
+  aaa_user:
+    description:
+    - The name of the locally-authenticated user user to add.
+    aliases: [ name, user ]
+  clear_password_history:
+    description:
+    - Whether to clear the password history of a locally-authenticated user.
+    type: bool
+  description:
+    description:
+    - Description for the AAA user.
+    aliases: [ descr ]
+  email:
+    description:
+    - The email address of the locally-authenticated user.
+  enabled:
+    description:
+    - The status of the locally-authenticated user account.
+    type: bool
+  expiration:
+    description:
+    - The expiration date of the locally-authenticated user account.
+  expires:
+    description:
+    - Whether to enable an expiration date for the locally-authenticated user account.
+    type: bool
+  first_name:
+    description:
+    - The first name of the locally-authenticated user.
+  last_name:
+    description:
+    - The last name of the locally-authenticated user.
+  phone:
+    description:
+    - The phone number of the locally-authenticated user.
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment: aci
+'''
+
+EXAMPLES = r'''
+- name: Add a user
+  aci_aaa_user:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    aaa_user: dag
+    aaa_password: AnotherSecretPassword
+    expiration: never
+    expires: no
+    email: dag@wieers.com
+    phone: 1-234-555-678
+    first_name: Dag
+    last_name: Wieers
+    state: present
+
+- name: Remove a user
+  aci_aaa_user:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    aaa_user: dag
+    state: absent
+
+- name: Query a user
+  aci_aaa_user:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    aaa_user: dag
+    state: query
+
+- name: Query all users
+  aci_aaa_user:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    state: query
+'''
+
+RETURN = r''' # '''
+
+from ansible.module_utils.network.aci.aci import ACIModule, aci_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        aaa_password=dict(type='str', no_log=True),
+        aaa_password_lifetime=dict(type='int'),
+        aaa_password_update_required=dict(type='bool'),
+        aaa_user=dict(type='str', required=True, aliases=['name']),
+        clear_password_history=dict(type='bool'),
+        description=dict(type='str', aliases=['descr']),
+        email=dict(type='str'),
+        enabled=dict(type='bool'),
+        expiration=dict(type='str'),
```

Comment:
what if we determined `expires` based on the presence of `expiration`? Also, how are users supposed to enter an expiration besides never? Should that be broken into component parts for clarity?

Reply:
Well I wanted to leave everything open as it is. Since the APIC allows combinations I didn't want to exclude any options. The documentation doesn't state the format, but it seems to be the ISO date format. To be tested/documented.

Reply:
So we accept now any python parseable datetime object. Which is pretty flexible.
And then we convert it to an ACI ISO8601-compatible string (which is non-standard on Python 2.7 :-/)